packages/frint/src/App.ts

```diff
@@ -116,6 +123,19 @@ export class App {
       throw new Error('Must provide `name` in options');
     }
 
+    // expose methods as class properties
+    Object.keys(this.options.methods).forEach(methodName => {
+      const method = this.options.methods[methodName];
+
+      if (typeof method === 'function') {
+        if (this[methodName] !== undefined) {
+          throw new Error(`Cannot overwrite app's \`${methodName}\` property or method with options method.`);
+        }
+
+        this[methodName] = method.bind(this);
```

Comment:
I don't know, i think we are doing some magic here, seems like a bad smell to me, we now are binding those methods and changing the scope, so we are exposing app instance inside each function, is that the desired behaviour?

What do you guys think?

const FooApp = createApp({
  name: 'Foo',
  methods: { // specify what methods this app exposes
    bar() {
      this.methods = {}; // exposing app variable
      console.log('bar')
    }
  }
})

Reply:
That's an interesting point, and I can see how this may give a lot of freedom to break things. On the other hand, if the framework doesn't automatically bind the methods, the developer could still do and have the same privileges, so I'm not sure if this is a big problem 🤔

Reply:
But there is another problems here too, if we need to create new methods on this class, and someone is using that name, when we update frintjs, this application is going to break too. We are also breaking the Open Close principle here i think.

Reply:
We are really open to modification on this case. 
But the thing is, imagine that i create a method called getMethods on my application, and then we do the same on App. Now when i update my frint package my application is going to break.

I think this is not a good thing :/

Reply:
One possible drawback I foresee is that, once this is merged, any new properties or methods added to the `App` class would require a major release, since they can potentially clash with injected methods. Should we worry about this, or is the `App` API stable enough for this to not be an issue?

Reply:
we haven't added any new methods for months, if not years to the App class.